mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -102,33 +151,80 @@ class Completions(
   inline private def undoBacktick(label: String): String =
     label.stripPrefix("`").stripSuffix("`")
 
-  private def findSuffix(methodSymbol: Symbol): Option[String] =
-    if shouldAddSnippet && methodSymbol.is(Flags.Method) then
-      lazy val extensionParam = methodSymbol.extensionParam
+  private def getParams(symbol: Symbol) =
+    lazy val extensionParam = symbol.extensionParam
+    if symbol.is(Flags.Extension) then
+      symbol.paramSymss.filter(
+        !_.contains(extensionParam)
+      )
+    else symbol.paramSymss
+
+  private def findSuffix(symbol: Symbol): Option[String] =
+
+    val bracketSuffix =
+      if shouldAddSnippet && noSquareBracketExists
+        && (isTypePosition || isNewPosition || isInstantiationOrMethodCallPos)
+        && (symbol.info.typeParams.nonEmpty
+          || (symbol.isAllOf(
+            Flags.JavaModule
+          ) && symbol.companionClass.typeParams.nonEmpty))
+      then "[$0]"
+      else ""
+
+    val bracesSuffix =
+      if shouldAddSnippet && symbol.is(
+          Flags.Method
+        ) && !isTypePosition
+      then
+        val paramss = getParams(symbol)
+        paramss match
+          case Nil => ""
+          case List(Nil) => "()"
+          case _ if config.isCompletionSnippetsEnabled =>
+            val onlyParameterless = paramss.forall(_.isEmpty)
+            lazy val onlyImplicitOrTypeParams = paramss.forall(
+              _.exists { sym =>
+                sym.isType || sym.is(Implicit) || sym.is(Given)
+              }
+            )
+            if onlyParameterless then "()" * paramss.length
+            else if onlyImplicitOrTypeParams then ""
+            else if bracketSuffix == "[$0]" then "()"
+            else "($0)"
+          case _ => ""
+        end match
+      else ""
+
+    val templateSuffix =
+      if shouldAddSnippet && isNewPosition
+        && (symbol.toString.startsWith(
+          "trait"
+        ) // trait A{ def doSomething: Int}
+        // object B{ new A@@}
```

Comment:
what the meaning of `@@`, I imagine this represents cursor position (based on how we write completionSuite in Metals), but it doesn't make sense.

Reply:
Why? 
```
object {
    new A{
      override def doSomething: Int = 2
    }
}

```

Reply:
I misunderstood in a same way.